### ISSUE TITLE
Expose WASM Hot Reload capabilities via `.props` file for consumption by `dotnet watch`

### ISSUE THREAD

**Issue by MackinnonBuck** [dotnet/runtime#97462]
The `dotnet watch` tool currently has a [hard-coded](https://github.com/dotnet/sdk/blob/6f9c74184babc859b096f445edcc3c2f1a52d5bb/src/BuiltInTools/dotnet-watch/HotReload/BlazorWebAssemblyDeltaApplier.cs#L16-L18) set of default WASM hot reload capabilities so that it can initialize before the WebAssembly runtime starts up.

We should consider having the runtime provide these capabilities, possibly by including them in a `.props` file so that they can get picked up by `dotnet watch`.

Context: https://github.com/dotnet/aspnetcore/issues/51507.

**Comment by MackinnonBuck** [comment]
cc @tmat

**Comment by pavelsavara** [comment]
duplicate of what @lewing ?

**Comment by maraf** [comment]
> duplicate of what [@lewing](https://github.com/lewing) ?

I don't think this is a duplicate afaik. This one tracks changes needed to happen on runtime side.

**Comment by lewing** [comment]
@maraf please update

**Comment by maraf** [comment]
Done in https://github.com/dotnet/aspnetcore/pull/60030
